core/app/models/spree/shipment.rb

```diff
@@ -199,24 +199,6 @@ def selected_shipping_rate_id=(id)
       self.save!
     end
 
-    # Determines the appropriate +state+ according to the following logic:
-    #
-    # canceled   if order is canceled
-    # pending    unless order is complete and +order.payment_state+ is +paid+
-    # shipped    if already shipped (ie. does not change the state)
-    # ready      all other cases
-    def determine_state(order)
```

Comment:
also, @richardnuno had the idea that we could just return the state of the shipment here. however upon further talk with him that would just be a 'fix for now' solution and wouldn't take care of cleaning out the problem fully.